languages/go/concepts/methods/about.md

```diff
@@ -1 +1,59 @@
-TODO: add information on methods concept
+A [method][methods] is a function with a special _receiver_ argument. The receiver appears in its own argument list between `func` keyword and the name of the method.
+
+```go
+func (receiver type) MethodName(parameters) (returnTypes){
+
+}
+```
+
+You can only define a method with a receiver whose type is defined in the same package as the method. The receiver can be either a struct type or a non-struct type as in the example below.
+
+```go
+package main
+
+import "fmt"
+
+type Name string
+
+func (s Name) Greetings() string {
+	return fmt.Sprintf("Welcome %s !", s)
+}
+
+func main() {
+	s := Name("Bronson")
+	fmt.Println(s.Greetings())
+}
+// Output: Welcome Bronson !
+```
+
+Methods with a value receiver operates on a copy of the value passed to it, meaning that any modification done to the receiver inside the method is not visible to the caller.
+
+You can declare methods with pointer receivers in order to modify the value to which the receiver points. Such modifications are visible to the caller or the method as well.
+
+```go
+package main
+
+import "fmt"
+
+type rect struct {
+	width, height int
+}
+
+func (r *rect) area() int {
+	return r.width * r.height
+}
+
+func main() {
+	r := rect{width: 10, height: 20}
+	fmt.Println("Area is: ", r.area())
+}
```

Comment:
This example is not assigning any new values to the struct fields from within the method. Could be done without a pointer receiver. Can we adjust this?

Reply:
Absolutely, my bad. Please check my last commit .

Reply:
Hmm, this still works the same if you remove the pointer receiver.

Here an example that would not work any more if you remove the receiver pointer:

```go
type rect struct {
	width, height int
}

func (r *rect) squareIt() {
	r.height = r.width
}

func main() {
	r := rect{width: 10, height: 20}
	fmt.Printf("Width: %d, Height: %d\n", r.width, r.height)
	// Output: Width: 10, Height: 20

	r.squareIt()
	fmt.Printf("Width: %d, Height: %d\n", r.width, r.height)
	// Output: Width: 10, Height: 10
}
```

Reply:
@tehsphinx oh, I see now. Thanks for helping out with this!